src/wasm/wasm-emscripten.cpp

```diff
@@ -602,13 +602,21 @@ struct EmJsWalker : public PostWalker<EmJsWalker> {
       Expression* first = nullptr;
       if (block && block->list.size() > 0) {
         first = block->list[0];
+        // skip a potential initial nop
+        if (first->is<Nop>()) {
```

Comment:
Yes, that pass will replace the initial set with a single nop. There won't be anything more here since there's just one constant in the whole function.